### PR TITLE
chore: configure renovate bot

### PR DESCRIPTION
**Pull request by maybeanerd**
This adds a renovate config. For the bot to do its work, you will still need to add it to the repository: https://github.com/apps/renovate

The config will resolve to NOT automerge for now, but whenever you want to activate it, it will use fast-forward merges

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+    "extends": [
+        "config:recommended",
+    
+        ":pinVersions",
+        ":automergeMinor",
```
Comment (maybeanerd) [inline renovate.json:7]: Depending on if you want any automerges (after passing tests), we could remove of keep this. IMO it's very nice to have, especially if you don't have automatic prod releases

File: renovate.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        ":rebaseStalePrs",
+    
+        "group:linters"
+      ],
+      "labels": ["dependencies"],
+      "automergeStrategy": "fast-forward",
+      "bumpVersion": "patch"
```
Comment (maybeanerd) [inline renovate.json:14]: I personally like this a lot - whenever it updates a dependency, it also updates the main package version

Although I think this is mainly useful for web, where every commit to main also is a release. Might make sense to not include this for you, If you cant every prod release to be at max a single version bump (either major,minor,or patch) and not a combination of potentially many

File: renovate.json (hunk 1 above)
Comment (maybeanerd) [reply renovate.json:7]: As of now, renovate will automerge path and minor version updates, and no matter if all pipelines pass, will _not_ automerge major version updates.

**Review by matthiasemde — APPROVED** [review]
LGTM💯

**Pull request merged**